packages/calcite-components/src/components/menu-item/menu-item.tsx

```diff
@@ -444,6 +444,13 @@ export class CalciteMenuItem implements LoadableComponent, T9nComponent, Localiz
         <div class={CSS.textContainer}>
           <span>{this.text}</span>
         </div>
+        {this.href && (this.topLevelMenuLayout === "vertical" || !this.isTopLevelItem) ? (
+          <calcite-icon
+            class={CSS.hoverHrefIcon}
+            icon={dir === "rtl" ? "arrow-left" : "arrow-right"}
```

Comment:
Does this need a `key` attr?

Reply:
Moved this to its own render method and added a key.